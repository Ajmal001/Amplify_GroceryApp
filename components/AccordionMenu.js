import React from "react";
import { TouchableOpacity , StyleSheet } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { Subheading, Text } from "react-native-paper";
import PropTypes from "prop-types";

const AccordionMenu = ({ text, expanded, handlePress }) => {
  return (
    <TouchableOpacity
      onPress={handlePress}
      style={[styles.rowAligned, styles.spaceBetween]}
    >
      <Text>{text}</Text>
      <Ionicons
        name={expanded ? "ios-arrow-up" : "ios-arrow-down"}
        size={24}
        color="grey"
      />
    </TouchableOpacity>
  );
};

export default AccordionMenu;

const styles = StyleSheet.create({
  rowAligned: {
    flexDirection: "row",
    marginLeft: 20,
  },
  spaceBetween: {
    justifyContent: "space-between",
  },
});

AccordionMenu.propTypes = {
  text: PropTypes.string.isRequired,
  expanded: PropTypes.bool.isRequired,
  handlePress: PropTypes.func.isRequired,
};
