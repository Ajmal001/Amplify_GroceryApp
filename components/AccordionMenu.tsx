import React from "react";
import { TouchableOpacity, StyleSheet, Text } from "react-native";
import { Ionicons } from "@expo/vector-icons";

interface Props {
  text: string;
  expanded: boolean;
  handlePress: () => void;
}

const AccordionMenu: React.FC<Props> = ({ text, expanded, handlePress }) => {
  return (
    <TouchableOpacity
      onPress={handlePress}
      style={[styles.rowAligned, styles.spaceBetween]}
    >
      <Text>{text}</Text>
      <Ionicons
        name={expanded ? "ios-arrow-up" : "ios-arrow-down"}
        size={24}
        color="grey"
      />
    </TouchableOpacity>
  );
};

export default AccordionMenu;

const styles = StyleSheet.create({
  rowAligned: {
    flexDirection: "row",
  },
  spaceBetween: {
    justifyContent: "space-between",
  },
});
