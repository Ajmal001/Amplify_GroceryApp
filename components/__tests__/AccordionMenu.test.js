import React from 'react';
import renderer from 'react-test-renderer';
import AccordionMenu from '../AccordionMenu';

jest.mock("@expo/vector-icons", () => {
    return {
        Ionicons: () => {
        return <></>;
      },
    };
  });

test('renders correctly', () => {
  const tree = renderer.create(<AccordionMenu
    text={''} 
    expanded={true}
    handlePress={(x) => x}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});