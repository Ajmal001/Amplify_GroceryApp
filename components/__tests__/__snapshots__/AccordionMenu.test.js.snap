// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    
  </Text>
</View>
`;
